proposals/propertyDependencies.md

```diff
@@ -0,0 +1,292 @@
+# JSON Schema: The `propertyDependencies` Keyword
+
+## Abstract
+
+The `propertyDependencies` keyword is a more friendly way to select between two
+or more schemas to validate an instance against than is currently supported by
+JSON Schema.
+
+## Status
+
+**Current Status**: PROPOSAL
+
+TODO: We should have a short standard blurb outlining the stages involved in a
+feature making its way to stable status.
+
+TODO: Link to a document that describes the proposal => stable process in
+detail.
+
+## Note to Readers
+
+The issues list for this document can be found at
+<https://github.com/json-schema-org/json-schema-spec/issues?q=is%3Aissue+propertydependencies>.
+
+For additional information, see <https://json-schema.org/>.
+
+To provide feedback, use this issue tracker or any of the communication methods
+listed on the homepage.
+
+## Table of Contents
+
+## Conventions and Terminology
+
+All conventions and terms used and defined by the [JSON Schema Core
+specification](../jsonschema-core.html) also apply to this document.
+
+## Overview
+
+### Problem Statement
+A common need in JSON Schema is to select between one schema or another to
+validate an instance based on the value of some property in the JSON instance.
+There are a several patterns people use to accomplish this, but they all have
+significant [problems](#problems).
+
+OpenAPI solves this problem with the `discriminator` keyword. However, their
+approach is more oriented toward code generation concerns, is poorly specified
+when it comes to validation, and is couple to OpenAPI concepts that don't exist
+is JSON Schema. Therefore, it's necessary to define something new rather than
+adopt `discriminator`.
+
+### Solution
+
+The `dependentSchemas` keyword is very close to what is needed except it checks
+for the presence of a property rather than it's value. The chosen solution is to
+build on that concept to solve this problem.
+
+```jsonschema
+{
+  "propertyDependencies": {
+    "foo": {
+      "aaa": { "$ref": "#/$defs/foo-aaa" }
+    }
+  }
+}
+```
+
+The validation result is equivalent to the following schema.
+
+```jsonschema
+{
+  "if": {
+    "properties": {
+      "foo": { "const": "aaa" }
+    },
+    "required": ["foo"]
+  },
+  "then": { "$ref": "#/$defs/foo-aaa" }
+}
+```
+
+### Limitations
+
+The problem of choosing an alternative based on a property value could apply for
+a value of any JSON type, but `propertyDependencies` only solves this problem
+when the value is a string. One of the main goals of this keyword is to define
+something that's intuitive enough and easy enough to use that people will
+actually use it rather than fallback to `oneOf` because it's simple. Achieving
+those goals means that some trade-offs need to be made. {{alternatives}} lists
+some alternatives that we considered.
+
+## A Vocabulary for Applying Subschemas
```

Comment:
Sorry, I forgot to explain why I changed that. This was a combination of a few things. I thought it was unnecessary for the "Proposal" content to be empty. The "Target for Change" content makes perfect sense in that space and we don't need the extra section.

After moving that content, my next thought was that the content was fragile. It's referencing a section number and its title in the spec. Either of those could change and that reference would be wrong. I wanted to both reduce that fragility and make it more explicit that the change is to add this keyword to a specific vocabulary. So, I changed it to reference the vocabulary URI the keyword is being added to. In theory that should be permanent and never change. To be helpful, I also linked to the section in the spec that describes the vocabulary. I added named anchor to use for this reference specifically to make it less fragile.

However, later I was trying to replace the `Proposal` heading with something that doesn't use the word "proposal". I couldn't think of anything at the time, but thought it would be interesting mimic the hierarchy of the spec. (Interestingly, that's also what they did [here](https://tc39.es/proposal-export-default-from/).) Of course, this goes against the anti-fragile goal I had for the previous change, but I had no other ideas at the time. What I probably should have done was bring back the "Target for Change" I had removed from before and used that heading instead. However, if you're ok with the word "Extension", I think I like that better. Let me know which you prefer and I'll drop the spec-matched hierarchy. I wasn't happy with the fragility of that anyway.